staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -508,11 +509,69 @@ func (w *watchCache) WaitUntilFreshAndList(ctx context.Context, resourceVersion
 	if err != nil {
 		return listResp{}, "", err
 	}
+	return w.list(ctx, resourceVersion, key, opts)
+}
+
+// NOTICE: Structure follows the shouldDelegateList function in
+// staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go
+func (w *watchCache) list(ctx context.Context, resourceVersion uint64, key string, opts storage.ListOptions) (resp listResp, index string, err error) {
+	switch opts.ResourceVersionMatch {
+	case metav1.ResourceVersionMatchExact:
+		return w.listExactRV(key, "", resourceVersion)
+	case metav1.ResourceVersionMatchNotOlderThan:
+	case "":
+		// Legacy exact match
+		if opts.Predicate.Limit > 0 && len(opts.ResourceVersion) > 0 && opts.ResourceVersion != "0" && opts.Predicate.Continue == "" {
+			return w.listExactRV(key, "", resourceVersion)
+		}
+		// Continue
+		if len(opts.Predicate.Continue) > 0 {
+			continueKey, continueRV, err := storage.DecodeContinue(opts.Predicate.Continue, key)
+			if err != nil {
+				return listResp{}, "", errors.NewBadRequest(fmt.Sprintf("invalid continue token: %v", err))
+			}
+			if continueRV > 0 {
+				return w.listExactRV(key, continueKey, uint64(continueRV))
+			} else {
+				// Continue with negative RV is a consistent read - already handled via waitUntilFreshAndBlock.
+				store, ok := w.store.(orderedLister)
```

Comment:
Why can't we call `listLatestRV` here?

Reply:
We can, we even did in previous draft. I was just worried about indices not supporting continueKey.

Reply:
Done